code/game/machinery/computer/launchpad_control.dm

```diff
@@ -53,6 +147,20 @@
 		return FALSE
 	return TRUE
 
+/// Performs checks on whether or not the launch pad can be used.
+/// Returns `null` if there are no errors, otherwise will return the error string.
+/obj/machinery/computer/launchpad/proc/teleport_checks(obj/machinery/launchpad/pad)
+	if(QDELETED(pad))
+		return "ERROR: Launchpad not responding. Check launchpad integrity."
+	if(!pad.isAvailable())
+		return "ERROR: Launchpad not operative. Make sure the launchpad is ready and powered."
+	if(pad.teleporting)
+		return "ERROR: Launchpad busy."
+	var/turf/pad_turf = get_turf(pad)
+	if(pad_turf && is_centcom_level(pad_turf.z))
+		return "ERROR: Launchpad not operative. Heavy area shielding makes teleporting impossible."
```

Comment:
perhaps add handling for bluespace anchors / other teleport blocks. Not sure if do_teleport actually allows "simulating" though.

Reply:
Not possible to simulate, for now copy pasting the check into `teleport_checks`, refactoring do_teleport to allow for checks I feel is outside the scope of this PR